src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,214 @@
+        $Object.PSObject.Properties["ftk_AccountType"].Value = $ftk_AccountType
+    } else {
+        $Object | Add-Member -MemberType NoteProperty -Name "ftk_AccountType" -Value $ftk_AccountType -Force
+    }
+    
+    return $Object
+}
+
+
+<# Main Function #>
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter()]
+        [string]
+        $ActualCost,
+
+        [Parameter()]
+        [string]
+        $outputFilePath,
+
+        [Parameter()]
+        [hashtable]
+        $columnMapping,
+
+        [Parameter()]
+        [bool]
+        $ExportAllColumns,
+
+        [Parameter()]
+        [switch]
+        $Force
+    )
+    
+    if (-not (Test-Path -Path $ActualCost)) {
+        Write-Error "Input file $ActualCost does not exist."
+        return
+    }
+    
+    $outputDirectory = [System.IO.Path]::GetDirectoryName($outputFilePath)
+    if (-not (Test-Path -Path $outputDirectory -PathType Container)) {
+        New-Item -Path $outputDirectory -ItemType Directory
+    }
+
+    # Import the CSV data
+    $csvData = Import-Csv -Path $ActualCost
+    
+    # Iterate over each row and apply transformations
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+        $object = Add-AccountType -Object $object
+
+        # Create a new object with the mapped column names
+        $newObject = @{}
+        foreach ($column in $object.PSObject.Properties.Name) {
+            if ($ExportAllColumns -or $column -in $columnMapping.Keys) {
+                $newColumnName = if ($columnMapping.ContainsKey($column)) { $columnMapping[$column] } else { $column }
+                $newObject[$newColumnName] = $object.$column
+            }
+        }
+
+        # Output the new object
+        [PSCustomObject]$newObject
+    }
+    
+    # Export the transformed data to the output CSV file
+    $transformedData | Export-Csv -Path $outputFilePath -NoTypeInformation -Encoding UTF8
+}
+
+
+
+
+<# Testing Framework #>
+# Define the number of rows and columns for testing and it will display an estimated processing time.
+if ($RunTest -and -not $Force) {
+    $host.UI.RawUI.ForegroundColor = "Yellow"
+    Write-Output "Running test..." 
+    .\Estimate-ProcessingTime.ps1 -rows $rows -columns $columns -columnMapping $columnMapping -ExportAllColumns $ExportAllColumns
+    $host.UI.RawUI.ForegroundColor = "White"
+    $confirmation = Read-Host "Do you want to continue? (Y/N)"
+    if ($confirmation -ne 'Y' -and $confirmation -ne 'y') {
+        $host.UI.RawUI.ForegroundColor = "Red"
+        Write-Output "Operation aborted by the user."
+        $host.UI.RawUI.ForegroundColor = "White"
+        return
+    }
+}
+
+# The actual function call
+# Call the functions from the function with the specified parameters
+ConvertTo-FinOpsSchema -ActualCost $ActualCost -outputFilePath $outputFilePath -columnMapping $columnMapping -ExportAllColumns $ExportAllColumns
```

Comment:
FYI - This file gets directly added to our module, so there shouldn't be an extra code outside of the functions. If this is for testing only, that's fine. Might want to add a `# TODO: Remove before committing` or something if that's the purpose.